### PR TITLE
makes moretrees game agnostic

### PR DESCRIPTION
**Pull request by wsor4035**
makes moretrees game agnostic

requires:  
* [x] https://github.com/mt-mods/xcompat/pull/19
* [x] ~~TODO: biomes api?~~

steps:  
* [x] boot minetest without crashing
* [x] handle biomes/nodes spawning on
* [x] balancing or anything else? (might not be needed)

occurrences of `default:*` not cared about
* everything in [`screenshot.lua`](https://github.com/mt-mods/moretrees/blob/670f7f4e6165b4c92e407a44b1e27c8ae55bfabe/screenshot.lua)
* default specific overrides in general, gated these to default in this pr
* everything not already handled in node_defs.lua
* surfaces in tree defs, as long as they have other nodes to be placed on
* some tools that have other counterparts that are comparable
* some surfaces, since they have some game compatible nodes in them

### CODE REVIEW

**Comment by wsor4035** [comment]
will require adding some more materials to xcompat

**Comment by wsor4035** [comment]
nodes are all handled

**Comment by wsor4035** [comment]
discovered while two legacy spawn defs has biomes, we dont honor that in the translate function, so thats not a concern at this time

File: init.lua [hunk 1]
```diff
@@ -15,29 +15,31 @@
...
-	description = "Wooden Planks"
-})
+	minetest.override_item("default:wood", {
+		description = "Wooden Planks"
+	})
 
-minetest.override_item("default:leaves", {
-	description = "Leaves"
-})
+	minetest.override_item(xcompat.materials.apple_leaves, {
```
Comment (Niklp09) [inline init.lua:31]: Useless change, is inside a `if get_modpath("default")`.

**Pull request merged**